src/System.Console/tests/ConsoleEncoding.cs

```diff
@@ -75,4 +77,166 @@ public void TestEncoding(string inputString)
+    {
+        NoSuchCodePage invalidEncoding = new NoSuchCodePage();
+        Assert.Throws<IOException>(() => Console.InputEncoding = invalidEncoding);
+        Assert.NotEqual(invalidEncoding, Console.InputEncoding);
+    }
+
+    [Fact]
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public void OutputEncoding_SetDefaultEncoding_Success()
+    {
+        RemoteInvoke(() =>
+        {
+            Encoding encoding = Encoding.GetEncoding(0);
+            Console.OutputEncoding = encoding;
+            Assert.Equal(encoding, Console.OutputEncoding);
+            Assert.Equal((uint)encoding.CodePage, GetConsoleOutputCP());
+
+            return SuccessExitCode;
+        }).Dispose();
+    }
+
+    [Fact]
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public void OutputEncoding_SetUnicodeEncoding_SilentlyIgnoredInternally()
+    {
+        RemoteInvoke(() =>
+        {
+            Encoding unicodeEncoding = Encoding.Unicode;
+            Encoding oldEncoding = Console.OutputEncoding;
+            Assert.NotEqual(unicodeEncoding.CodePage, oldEncoding.CodePage);
+            Console.OutputEncoding = unicodeEncoding;
+            Assert.Equal(unicodeEncoding, Console.OutputEncoding);
+
+            Assert.Equal((uint)oldEncoding.CodePage, GetConsoleOutputCP());
+
+            return SuccessExitCode;
+        }).Dispose();
+    }
+
+    [Fact]
+    public void OutputEncoding_SetWithErrorAndOutputInitialized_ResetsErrorAndOutput()
+    {
+        RemoteInvoke(() =>
+        {
+            // Initialize Console.Error
+            TextWriter errorWriter = Console.Error;
+            Assert.NotNull(errorWriter);
+            Assert.Same(errorWriter, Console.Error);
+
+            // Initialize Console.Out
+            TextWriter outWriter = Console.Out;
+            Assert.NotNull(outWriter);
+            Assert.Same(outWriter, Console.Out);
+
+            // Chang the OutputEncoding
+            Console.OutputEncoding = Encoding.ASCII;
+            Assert.Equal(Encoding.ASCII, Console.OutputEncoding);
+
+            Assert.NotSame(errorWriter, Console.Error);
+            Assert.NotSame(outWriter, Console.Out);
+
+            return SuccessExitCode;
+        }).Dispose();
+    }
+
+    [Fact]
+    public void OutputEncoding_SetNull_ThrowsArgumentNullException()
+    {
+        Assert.Throws<ArgumentNullException>("value", () => Console.OutputEncoding = null);
+    }
+
+    [Fact]
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public void OutputEncoding_SetEncodingWithInvalidCodePage_ThrowsIOException()
+    {
+        NoSuchCodePage invalidEncoding = new NoSuchCodePage();
+        Assert.Throws<IOException>(() => Console.OutputEncoding = invalidEncoding);
+        Assert.NotEqual(invalidEncoding, Console.OutputEncoding);
+    }
+
+    [DllImport("kernel32.dll")]
+    public extern static uint GetConsoleCP();
+
+    [DllImport("kernel32.dll")]
+    public extern static uint GetConsoleOutputCP();
```

Comment:
These may need to be separated out into a Windows-specific file for these tests to run on UWP.